project/src/pages/main-page-screen/main-page-screen.tsx

```diff
@@ -3,16 +3,26 @@ import OffersList from '../../components/offers-list/offers-list';
 import { Helmet } from 'react-helmet-async';
 import { Offer } from '../../types/offer';
 import Map from '../../components/map/map';
-import { offers, location } from '../../mock/offers';
-import { useState } from 'react';
+import { offers } from '../../mock/offers';
+import { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { setListOffers } from '../../store/action';
+import CitiesList from '../../components/cities-list/cities-list';
 
-type Props = {
-  offersList: Offer[];
-}
-
-export default function MainPage({offersList}: Props) {
+export default function MainPage() {
   const [selectedOffer, setSelectedOffer] = useState<Offer | undefined>(undefined);
 
+  const dispatch = useAppDispatch();
+  const currentCity = useAppSelector((state) => state.city);
+  const allOffers = useAppSelector((state) => state.offers);
+  const currentOffers = useAppSelector(() => allOffers.filter((offer) => offer.city.name === currentCity.name));
+
+  useEffect(() => {
+    dispatch(setListOffers(
+      allOffers
+    ));
+  }, [dispatch, allOffers]);
+
```

Comment:
этот эффект тут не нужен,
вместо него другой напиши 
```suggestion
useEffect(() => {
    dispatch(setListOffers(offers));
  }, []);

```